cmd/juju/application/deployer/charm.go

```diff
@@ -372,6 +376,37 @@ func (c *repositoryCharm) String() string {
 // PrepareAndDeploy finishes preparing to deploy a repository charm,
 // then deploys it.
 func (c *repositoryCharm) PrepareAndDeploy(ctx *cmd.Context, deployAPI DeployerAPI, resolver Resolver) error {
+	if featureflag.Enabled(feature.ServerSideCharmDeploy) && deployAPI.BestFacadeVersion("Application") >= 18 {
```

Comment:
Is there a better way of doing this without having an if block like this? Since we'll need one in each `PrepareAndDeploy` for local and predeployed charms as well. All the ways I can think of create extra development overhead, so I guess this is the simplest way. Maybe we can find a good pattern for this in the future.